### DISCUSSION TITLE
Project Translation

### DISCUSSION THREAD

**Discussion by shuvashish76** [FunkyMuse/Vigilante#410] [Ideas]
I've seen OSS projects use these sites for translation purpose
- [Crowdin](https://crowdin.com/)
- [Transifex](https://www.transifex.com/)
- [Weblate](https://weblate.org/en/)

I think Weblate is better as it's [opensource](https://github.com/WeblateOrg/weblate).

Features: https://weblate.org/en/features/
![Screenshot_2021-12-05-21-15-09](https://user-images.githubusercontent.com/48226312/144753433-4fc61e84-f12d-42d0-885e-1b9abd399061.jpeg)
https://weblate.org/en/hosting/

---

https://github.com/FunkyMuse/Vigilante#translations

> To add a translation, please open a pull request, Crowdin was removed because it worked 2 out of 10 times.

I see 👍🏿

**Comment by FunkyMuse** [comment]
[Crowdin](https://crowdin.com/) was removed because it worked 1/12 of the time

**Comment by shuvashish76** [comment]
Ping @comradekingu Can you please help for Weblate integration?

**Reply by comradekingu** [reply]
@shuvashish76 Yes. Sorry, I thought I answered earlier.
Edit: I got something up at https://hosted.weblate.org/projects/vigilante/
With a yolomerge of #420 and #421 it may hold if https://docs.weblate.org/en/latest/admin/continuous.html?highlight=webhooks#automatically-receiving-changes-from-github is added and possibly the https://github.com/weblate user added to the org.
Edit2: I got a nb_NO translation in. The app strings from that effort showed up as https://github.com/FunkyMuse/Vigilante/pull/422

**Reply by comradekingu** [reply]
Sent a request for permanent hosting of the configured translation effort now.

**Comment by comradekingu** [comment]
@shuvashish76 @FunkyMuse 
Do you have accounts I can add as Weblate admins, so it isn't just me able to change things?

**Reply by FunkyMuse** [reply]
> @shuvashish76 @FunkyMuse Do you have accounts I can add as Weblate admins, so it isn't just me able to change things?

Yes, FunkyMuse is my username there too

**Reply by comradekingu** [reply]
Added. I think that concludes the setup.
There are a few more stringfixes to be had, which I will get around to PR.
Edit: https://github.com/FunkyMuse/Vigilante/pull/488

**Reply by FunkyMuse** [reply]
Thanks buddy

**Comment by sudo-py-dev** [comment]


**Reply by FunkyMuse** [reply]
I'm too busy to push an update soon, maybe end of this or next week i will.

**Comment by sudo-py-dev** [comment]


**Reply by FunkyMuse** [reply]
Friendly reminder, stop spamming @Shopimisrel when an update comes out, you'll see it.

**Comment by Porfik** [comment]
I made a full translation into Russian, can you commit it?

**Reply by comradekingu** [reply]
@Porfik
Top stuff :)
The changes have already made their way in https://github.com/FunkyMuse/Vigilante/pull/601/files
It will all work if you have selected Russian as the language used in Android for when the next release comes out.
Provided the prior release cadence of 3-6 months holds up, the last release was 2.5 months ago.
It is possible to donate to https://liberapay.com/crazylegend/ in the hopes of it happening sooner rather than later I suppose.
